src/metabase/models/interface.clj

```diff
@@ -178,21 +182,20 @@
     (post-select <>)
     (map-> entity <>)))
 
-
 (def IEntityDefaults
   "Default implementations for `IEntity` methods."
-  {:default-fields       (constantly nil)
-   :timestamped?         (constantly false)
-   :hydration-keys       (constantly nil)
-   :types                (constantly nil)
-   :can-read?            (constantly true)
-   :can-write?           superuser?
-   :pre-insert           identity
-   :post-insert          identity
-   :pre-update           identity
-   :post-update          (constantly nil)
-   :post-select          identity
-   :pre-cascade-delete   (constantly nil)})
+  {:default-fields            (constantly nil)
+   :timestamped?              (constantly false)
+   :hydration-keys            (constantly nil)
+   :types                     (constantly nil)
+   :can-read?                 (constantly true)
+   :can-write?                superuser?
```

Comment:
That's a little obscure for anyone who doesn't know how our permissions checking works. I'll make the default implementation throw an `UnsupportedOperationException` with instructions on what needs to be done